experimental/Matrix/matrix.jl

```diff
@@ -0,0 +1,107 @@
+module MatrixGroups
+
+using GAP
+using Oscar
+
+export _wrap_for_gap
+  
+################################################################################
+# Initilize GAP function, i.e. GAP reads the file matrix.g
+#
+function __init__()
+    GAP.Globals.Reread(GAP.GapObj(joinpath(Oscar.oscardir, "experimental", "Matrix", "matrix.g")))
+end
+
+################################################################################
+# Computes the JuliaMatrixRep of a MatrixElem in GAP
+#
+#    _wrap_for_gap(m::MatrixElem)
+#
+# Input: m::MatrixElem
+# Output: Returns the JuliaMatrixRep of m in GAP
+#
+# Example
+# ```jldoctest
+# julia> m = matrix(ZZ, [0 1 ; -1 0]);
+# julia> _wrap_for_gap(m)
+# GAP: <matrix object of dimensions 2x2 over Integer Ring>
+# ```
+# """
+function _wrap_for_gap(m::MatrixElem)
+    return GAP.Globals.MakeJuliaMatrixRep(m)
+end
+
+
+################################################################################
+# Computes a GAP Group where the generators are the JuliaMatrixRep of matrices.
+# Moreover, a nice morphism from G into a GAP matrix group over a finite field G2 is constructed
+# such that calculations in G can be handled by G2 automatically.
+#
+#     MatrixGroup(matrices::Vector{<:MatrixElem{T}}) where T <: Union{fmpz, fmpq, nf_elem}
+#
+# Input: matrices::Vector{<:MatrixElem{T}} where T <: Union{fmpz, fmpq, nf_elem}
+# Output: GAP group generated by the JuliaMatrixReps of matrices
+#
+# Example
+# ```jldoctest
+# julia> m1 = matrix(QQ, [0 1 ; -1 0]);
+# julia> m2 = matrix(QQ, [ -1 0; 0 1]);
+# julia> MatrixGroup([m1,m2])
+# GAP: <group with 2 generators>
+# ```
+# """
+function MatrixGroup(matrices::Vector{<:MatrixElem{T}}) where T <: Union{fmpz, fmpq, nf_elem}
+       @assert !isempty(matrices)
+    
+       # One should probably check whether all matrices are n by n (and invertible
+       # and such ...)
```

Comment:
Indeed, and that they all have the same base_ring, too

Reply:
I have added some checks. Am I missing something?